incydr/types.py

```diff
@@ -0,0 +1,38 @@
+import click
+
+
+# TODO: do we want this ?
```

Comment:
This `AutoDecodedFile` class is the original parent class for `FileOrString`.  Do we know what the benefit of detecting encoding before `click.File` does is?

Reply:
the reasoning for adding the encoding check is detailed here: https://github.com/code42/code42cli/pull/265